llvm/include/llvm/Support/FileSystem.h

```diff
@@ -1200,7 +1200,10 @@ std::error_code getRealPathFromHandle(file_t Handle,
 /// Lock the file.
 ///
 /// This function acts as @ref tryLockFile but it waits infinitely.
-std::error_code lockFile(int FD);
+/// \param FD file descriptor to use for locking.
+/// \param Exclusive if \p true use exclusive/writer lock, otherwise use
+/// shared/reader lock.
+std::error_code lockFile(int FD, bool Exclusive = true);
```

Comment:
This should get tested, maybe in `llvm/unittests/Support/Path.cpp`.  In fact, maybe we should just upstream this?

Reply:
> maybe we should just upstream this?

Yes, and I'd like to postpone looking into what test is appropriate for this until the upstream patch.